src/deck/guides/3.0-upgrade.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Upgrade to Kong Gateway 3.x
+---
+
+Starting with {{site.base_gateway}} 3.0, paths using regular expressions must have a `~` prefix.
+The decK format version must be `3.0` to work with the new paths.
+
+If you [migrate your {{site.base_gateway}} database](/gateway/latest/upgrade/) from 2.8.x to 3.0, the paths in the database will have the `~` prefix added.
+This can cause configuration drift between state files and the database.
+
+Before using your state files to update the database, convert them into 3.0 format using `deck convert`.
+
+{:.important}
+> **Important**: Do not use `deck sync` with {{site.base_gateway}} 3.x before converting paths into 3.0 format.
+This will break all regex routing in 3.x.
+
+## decK command behavior with 3.x
+
+When running decK commands against {{site.base_gateway}} 3.x, take note of the following behavior:
+
+`deck dump`
+: Explicitly sets the format version to 3.0.
+It assumes that the paths have been correctly transformed, either via Kong migrations, manually, or through a `deck convert`.
+
+`deck diff`, `deck validate`, or `deck sync`
+: decK performs a check to ensure all regex routes are correctly prefixed.
+If not, the behavior depends on the control plane type:
+* Self-managed {{site.base_gateway}} 2.x (format version 1.1 or earlier): Prints an error and stops the operation.
```

Comment:
Would it be better to be as explicit as possible and use `_format_version` here? WDYT?